moo.js

```diff
@@ -407,10 +407,10 @@
     this.setState(state)
   }
 
-  Lexer.prototype._eat = hasSticky ? function(re) { // assume re is /y
-    return re.exec(this.buffer)
-  } : function(re) { // assume re is /g
-    var match = re.exec(this.buffer)
+  Lexer._eat = hasSticky ? function(re, buffer) { // assume re is /y
+    return re.exec(buffer)
+  } : function(re, buffer) { // assume re is /g
+    var match = re.exec(buffer)
```

Comment:
This should save a lookup on `this`.

Reply:
If you're going to do this, why not just do `var eat = …`?